storage/src/lib.rs

```diff
@@ -330,14 +330,15 @@ impl Storage {
     }
 
     pub fn drop_loose_index_before(&mut self, diff: ChainDifficulty) {
-        if self.loose_idx.is_empty() {
-            return;
-        }
-        let drop_diff = u64::from(diff);
-        let tip_diff = u64::from(self.loose_idx[0].0);
-        let drop_after = (tip_diff - drop_diff) as usize;
-        if self.loose_idx.len() > drop_after {
-            self.loose_idx = self.loose_idx[0..drop_after].to_vec();
+        if let Some(&(idx_tip_diff, _, _)) = self.loose_idx.first() {
```

Comment:
I wonder if it's good to have non-descriptive three item tuples there, maybe rework them into proper structs?

Reply:
Makes sense, yeah 👌